### PR TITLE
Add partial Ed25519 support

### PR DESCRIPTION
**Pull request by trishankatdatadog**
**Fixes issue #**:

Presently, in-toto supports using only RSA keys.

**Description of the changes being introduced by the pull request**:

This PR supports using both RSA and Ed25519 keys.

**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

### CODE REVIEW

File: in_toto/util.py [hunk 1]
```diff
@@ -134,8 +149,39 @@ def prompt_password(prompt="Enter password: "):
   return getpass.getpass(prompt, sys.stderr)
 
 
+def prompt_import_private_key_from_file(filepath):
+  """Trys to load an Ed25519 or RSA key without password. If a CryptoError
+  occurs, prompts the user for a password and trys to load the the key again.
+  """
+  # FIXME: The more secure method is to load the private key using the key type
```
Comment (trishankatdatadog) [inline in_toto/util.py:156]: This function is dangerous in the sense that it makes the assumption that a private key is always either an Ed25519, or an RSA key, in that order. I haven't fully thought this through, but an attacker might be able to exploit this in order to bind private keys to [mismatching cryptographic schemes](https://www.nccgroup.trust/us/our-research/docker-notary/).

A more robust function would either: (1) ask the user to explicitly identify the keytype, and / or (2) correctly guess the keytype from information in the private key itself.

**Comment by trishankatdatadog** [comment]
Related to https://github.com/secure-systems-lab/securesystemslib/pull/148

**Comment by trishankatdatadog** [comment]
It looks like the tests are failing because of some fixes for GPG that is on [add-substitution-layer](https://github.com/in-toto/in-toto/tree/add-substitution-layer) but not [develop](https://github.com/in-toto/in-toto/tree/develop).

Update: huh, I spoke too soon. For whatever reason, tox fails on my machine, but works on coveralls. Who the hells knows what's happening.

**Comment by trishankatdatadog** [comment]
This PR is certainly not complete, because the other CLI tools need to be patched and tested, too.

File: tox.ini [hunk 2]
```diff
@@ -15,6 +15,7 @@ deps =
     bandit
 
 commands =
+    pip install -U https://github.com/secure-systems-lab/securesystemslib/archive/master.zip
```
Comment (trishankatdatadog) [inline tox.ini:18]: Also, please be sure to remove this line --- it's only there for testing.

**Comment by vladimir-v-diaz** [comment]
Note: I reviewed pull request #209, which is based on and expands this pull request.

**Comment by SantiagoTorres** [comment]
I merged #209, which supersedes this PR. I'll close this one now.

**Pull request closed without merging**